QuantumToolbox.jl/time_evolution/adiabatic.qmd

```diff
@@ -0,0 +1,179 @@
+
+For this theorem to hold, certain conditions must be satisfied: the system must evolve slow enough compared to the energy gap between the relevant eigenstate and other energy levels, and this gap must remain non-zero throughout the evolution.
+
+Here, we will demonstrate the well-known application of the adiabatic theorem in quantum computing called $\text{\emph{adiabatic sweep}}$. This is a method for preparing some desired quantum state by slowly evolving a quantum system with a time-dependent Hamiltonian. Essentially, the adiabatic theorem allows us to prepare the ground state of the final Hamiltonian $H_1$ from the ground state of the initial Hamiltonian $H_0$.
+
+## Model
+
+We consider a chain of $N$-identical spin-$1/2$ particles to study their spin dynamics and set our interest in finding the ground state of the condition that the spin chain has some random gap, leading to the random magnitude $g_i$ of $\hat{\sigma}^i_x \hat{\sigma}^{i+1}_x$ interaction with the neighboring spin.
+
+Initially, we prepare the system such that the spins are free from interaction with each other and are all in the ground state, i.e.,
+
+$$H_0 = \sum_i^N \frac{\varepsilon_0}{2} \hat{\sigma}^i_z,$$
+$$|\psi(0)\rangle = \bigotimes_{i=1}^N |g\rangle$$
+
+Then, gradually, the Hamiltonian evolves to:
+
+$$H_1 = \sum_{i=1}^N \frac{\varepsilon_0}{2} \hat{\sigma}^i_z + \sum_{i=1}^{N-1} g_i \hat{\sigma}^i_x \hat{\sigma}^{i+1}_x,$$
+
+whose ground state are desired. By gradual change, we are subject to the simplest form of adiabatic sweep, i.e.,
+
+$$H(t,T) = H_0 * (1-t/T) + H_1 * t/T,$$
+
+where the parameter $T$ determines how slow the Hamiltonian changes in time.
+
+## Code demonstration
+
+```{julia}
+using QuantumToolbox
+using CairoMakie
+```
+
+```{julia}
+N = 8 # number of spins
+ε0 = 1 # energy gap of the spins
+
+gs = rand(N-1) # use rand function for the random interaction strengths
+```
+
+```{julia}
+Is = Vector{QuantumObject}(fill(qeye(2), N));
+
+H0 = QuantumObject(
+    zeros((2^N, 2^N)),
+    Operator,
+    Tuple(fill(2, N))
+)
+
+for idx in 1:N
+    is = copy(Is)
+    is[idx] = ε0/2 * sigmaz()
+    H0 += kron(is...)
+end
+
+ψ0 = kron(fill(basis(2,1), N)...)
+print(H0)
+```
+
+```{julia}
+H1 = QuantumObject(
+    zeros((2^N, 2^N)),
+    Operator,
+    Tuple(fill(2, N))
+)
+
+for idx in 1:N
+    is = copy(Is)
+    is[idx] = ε0/2 * sigmaz()
+    H1 += kron(is...)
+end
+
+for idx in 1:N-1
+    is = copy(Is)
+    is[idx] = gs[idx] * sigmax()
+    is[idx+1] = sigmax()
+
+    h_int = kron(is...)
+        
+    H1 += h_int
+end
+
+print(H1)
+```
+
+Here, we define the function `H(T)` where `T` is the total evolution time.
+```{julia}
+H(T) = QuantumObjectEvolution((
+    (H0, (p,t) -> 1 - t/T),  # the dummy parameter `p` is not used but still required by the ODE solver
+    (H1, (p,t) -> t/T),
+))
```

Comment:
Instead of put it as a function, you can define it using the parameters

```suggestion
H = QuantumObjectEvolution((
     (H0, (p,t) -> 1 - t/p.T), 
     (H1, (p,t) -> t/p.T),
 ))
```

So that we can show the potentiality of `QuantumObjectEvolution`